### PR TITLE
Olga hw 38

### PR DESCRIPTION
**Pull request by OlechkaM**

### CODE REVIEW

**Comment by OlechkaM** [comment]
I did something wrong and my ButtonClass doesn't work. Can you check it, please?

File: my-app/src/Button/ButtonClass.tsx [hunk 1]
```diff
@@ -1,29 +1,43 @@
...
-        return (
-            <button className={styles.button} onClick={this.click}>{this.state.opened ? <Burger /> : <Close />}  </button>
-        )
+    const click = () => {
+        handleClick()
+        setIsVisible(!isVisible)
     }
 
+    return (
+        <button className={styles.button} onClick={click}>{visible ? <Burger /> : <Close />}  </button>
```
Comment (jathpr) [inline my-app/src/Button/ButtonClass.tsx:18]: you check prop visible, that stay same all time
you need to check state

File: my-app/src/hw-38/PostsList.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import { ReactNode } from 'react';
```
Comment (jathpr) [inline my-app/src/hw-38/PostsList.tsx:1]: you do not use `ReactNode` here

File: my-app/src/hw-38/PostsList.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
```
Comment (jathpr) [inline my-app/src/hw-38/PostsList.tsx:2]: rename file `postTypes.ts`
main parts - start from lower case
extension - ts

**Review by jathpr — APPROVED** [review]
nice

**Comment by jathpr** [comment]
can be merged

File: my-app/src/Button/ButtonClass.tsx (hunk 1 above)
Comment (OlechkaM) [reply my-app/src/Button/ButtonClass.tsx:18]: i fixed it

**Pull request merged**